common/web/lm-worker/src/main/models/dummy-model.ts

```diff
@@ -30,15 +30,33 @@
  * prediction API exclusively.
  */
 
+interface Wordform2Key {
+  (wordform: string): string;
+}
+
+export interface DummyOptions {
+  toKey?: Wordform2Key,
+  futureSuggestions?: Outcome<Suggestion>[][];
+  punctuation?: LexicalModelPunctuation;
+  applyCasing?: (form: CasingForm, text: string) => string;
+  wordbreaker?: WordBreakingFunction;
+  languageUsesCasing?: boolean;
+}
+
 /**
  * The Dummy Model that returns nonsensical, but predictable results.
  */
-export default class DummyModel implements LexicalModel {
+export class DummyModel implements LexicalModel {
   configuration: Configuration;
   punctuation?: LexicalModelPunctuation;
-  private _futureSuggestions: Suggestion[][];
+  toKey?: Wordform2Key;
+  applyCasing?: (form: CasingForm, text: string) => string;
+  wordbreaker?: WordBreakingFunction;
+  languageUsesCasing?: boolean;
```

Comment:
The dummy model doesn't actually use these internally, of course.  The component methods for the prediction process, however, do use them - and so this makes a great spot to mock the requirements for those methods.